src/state.rs

```diff
@@ -190,38 +190,50 @@ impl StateWrite {
     Ok(())
   }
 
-  pub fn commit(&mut self, repo: &Repo, data: CommitArgs) -> Result<()> {
+  pub fn commit(&mut self, repo: &Repo, args: CommitArgs) -> Result<()> {
+    for proj_id in &self.proj_writes {
+      if let Some((root, hooks)) = args.hooks.get(proj_id) {
+        hooks.execute_pre_begin(root)?;
+      }
+    }
+
+    for proj_id in &self.proj_writes {
+      if let Some((root, hooks)) = args.hooks.get(proj_id) {
+        hooks.execute_pre_write(root)?;
+      }
+    }
```

Comment:
Do we really need two hooks that are in the same place but different name?

Reply:
It does seem a little weird, but the idea is that the hooks are semantically different. Long term goal would be that the pre-begin hook runs on all projects on every release, whereas the pre-write hook runs only on projects that actually will write something. Also, if there later exists some hook-able action that occurs before write, the pre-write hook will come after that, but pre-begin will still come before it. Finally, because hooks are all run together, this will give a user to have some control over when their hooks will run in relation to each other.